nixos/modules/services/networking/networkmanager.nix

```diff
@@ -461,6 +461,8 @@ in {
       "d /var/lib/NetworkManager-fortisslvpn 0700 root root -"
 
       "d /var/lib/misc 0755 root root -" # for dnsmasq.leases
+      # ppp isn't able to mkdir that directory at runtime
+      "d /run/pppd/lock 0700 root root -"
```

Comment:
There's a pppd module, so this should probably be there? https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/pppd.nix